### PR TITLE
Dev nap

### PR DESCRIPTION
**Pull request by ArtNattapat**
change NAP.config to require('./config')
move nap's graphql to server/graphql
graphql can serve only 1 endpoint now (from multi endpoint by sub-directory)
extend support by global.NAP.expose (see graphq/setup.example.js for more example)

### CODE REVIEW

File: graphql/setup.example.js [hunk 1]
```diff
@@ -0,0 +1,19 @@
+global.NAP.expose.extendModel('User', {
+  extenddedString: 'String',
+});
+
+global.NAP.expose.setBuildGraphqlSchema(({ GQC, models }) => {
+  models.UserTC.addFields({
+    extendedFeild: {
+      type: 'String',
+      resolve: () => 'HEllo',
```
Comment (katopz) [inline graphql/setup.example.js]: Please correct `HEllo` to `Hello`

File: graphql/setup.example.js [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+    },
+  });
+  GQC.rootQuery().addFields({
+    sss: {
```
Comment (katopz) [inline graphql/setup.example.js]: Please change `sss` to `foo`

File: examples/module/index.js [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+nap.start({
+  buildGraphqlSchema: ({ GQC, models }) => {
+    models.UserTC.addFields({
+      extendedFeild: {
+        type: 'String',
+        resolve: () => 'HEllo',
+      },
+    });
+    GQC.rootQuery().addFields({
+      sss: {
```
Comment (katopz) [inline examples/module/index.js]: Please change `sss` to `foo`

File: lib/nap-plugin-oauth/README.md [hunk 4]
```diff
@@ -0,0 +1,20 @@
+
+## Usage
+
+```ts
```
Comment (katopz) [inline lib/nap-plugin-oauth/README.md]: Please change `ts` to `js`

File: server/config.js [hunk 5]
```diff
@@ -20,4 +19,12 @@ global.NAP.Config = {
   // Security
   cookie_secret: process.env.COOKIE_SECRET || 'foo',
   jwt_secret: process.env.JWT_SECRET || 'foo',
-}
\ No newline at end of file
+  
+  passportEnabled: int(process.env.PASSPORT_DISABLED || '0') === 0,
```
Comment (katopz) [inline server/config.js]: `Passport` is opt-out service so please use `passport_disabled` instead

File: server/config.js [hunk 6]
```diff
@@ -20,4 +19,12 @@ global.NAP.Config = {
...
+  graphqlEnabled: true,
```
Comment (katopz) [inline server/config.js]: `GraphQL` is opt-out service so please use `graphql_disabled` instead

File: server/config.js [hunk 7]
```diff
@@ -20,4 +19,12 @@ global.NAP.Config = {
...
+  graphqliqlEnabled: true,
```
Comment (katopz) [inline server/config.js]: `GraphiQL` should enable by default in dev mode and opt-in at prod please use `graphiql_enabled` and respect `process.env.GRAPHIQL_ENABLED`

File: server/config.js [hunk 8]
```diff
@@ -20,4 +19,12 @@ global.NAP.Config = {
...
+
+  mubsub: process.env.TRIGGER_MUBSUB,
```
Comment (katopz) [inline server/config.js]: `mubsub` should be optional and opt-in

File: server/index.js [hunk 9]
```diff
@@ -1,16 +1,16 @@
...
 
+const start = () => {
   // Next
+  global.NAP = {};
   const nextjs = require('next')({ dev: config.dev })
 
   // Will apply middleware
   const initializer = require('./initializer')
-  return nextjs.prepare().then(() => initializer(config, nextjs))
+  return nextjs.prepare().then(() => initializer(config, nextjs)).catch(console.log)
```
Comment (katopz) [inline server/index.js]: Please remove `console.log`

File: server/initializer.js [hunk 10]
```diff
@@ -11,21 +11,24 @@ module.exports = async (config, nextjs) => {
...
   app.use(express.static('public'))
 
+  require('./initMubsub')()
+
   // Mongoose
   const mongoose = await require('./initMongoose')(config.mongo_url)
 
   // Passport
-  process.env.PASSPORT_DISABLED !== '1' && require('./initPassport')(config, app)
+  config.passportEnabled && require('./initPassport')(config, app, nextjs)
```
Comment (katopz) [inline server/initializer.js]: Please replace `(config, app, nextjs)` with `(config, app)`

File: templates/email-register.js [hunk 11]
```diff
@@ -0,0 +1,10 @@
+const config = require('../server/config');
+
+// Will send this to sign in with email user via MailGun
+// @see https://github.com/mailgun/mailgun-js#messages
+module.exports = (email, verification_url) => ({
+  from: 'noreply@' + config.MAILGUN_DOMAIN,
+  to: [email],
+  subject: 'Hello from Mailgun',
+  text: 'Use the link below to sign in:\n\n' + verification_url + '\n\n'
+})
```
Comment (katopz) [inline templates/email-register.js]: This file should be delete (deprecated)

**Review by katopz — CHANGES_REQUESTED** [review]

File: examples/module/index.js (hunk 3 above)
Comment (ArtNattapat) [reply examples/module/index.js]: this file deprecated. i will remove on next commit.

File: lib/nap-plugin-oauth/README.md (hunk 4 above)
Comment (ArtNattapat) [reply lib/nap-plugin-oauth/README.md]: this example written by typescript. should it be ts format ?

Comment (katopz) [reply lib/nap-plugin-oauth/README.md]: Please also provide `js` example.

File: server/initializer.js [hunk 12]
```diff
@@ -11,16 +11,16 @@ module.exports = async (config, nextjs) => {
...
 
-  require('./initMubsub')()
+  config.mubsub_enabled && require('./initMubsub')()
 
   // Mongoose
   const mongoose = await require('./initMongoose')(config.mongo_url)
 
   // Passport
-  config.passportEnabled && require('./initPassport')(config, app, nextjs)
+  !config.passport_disabled && require('./initPassport')(config, app, nextjs)
```
Comment (katopz) [inline server/initializer.js]: Please change `(config, app, nextjs)` to `(config, app)`

**Review by katopz — CHANGES_REQUESTED** [review]

**Review by katopz — APPROVED** [review]

**Pull request merged**